### DISCUSSION TITLE
Vue files support ?

### DISCUSSION THREAD

**Discussion by vilanele** [hedyhli/outline.nvim#69] [Help]
Hi, thanks for this great plugin.
Currently nothing is showing up in the outline window for Vue files. I'm using lsp volar for Vue files.
Am I missing something or Vue files are not supported yet ?

**Comment by hedyhli** [comment]
Hey there, support for certain filetypes for LSP depends entirely on whether the LSP in question supports a specific capability, namely `documentSymbolProvider`.

If Volar supports it, then outline will work; if it doesn't, then there's not much we can do on our side to make it work, because we are fetching symbols from the LSP, which is Volar here.

I don't use Vue personally but I can help look into whether Volar supports document symbols.

Where can I install Volar? A quick search yields a number of results of different projects 😅

**Comment by vilanele** [comment]
Hello,
The proper setup is : 

1. install the typescript lsp "ts_ls": `npm install -g typescript typescript-language-server`
2. install the vue plugin for ts_ls: `npm -g install @vue/typescript-plugin`
3. Configure ts_ls properly as in the section "Vue support" of ts_ls in https://github.com/neovim/nvim-lspconfig/blob/master/doc/configs.md#ts_ls. Just change the location field to the right install dir of @vue/typescript-plugin.
4. Install volar lsp : `npm install -g @vue/language-server`
5. Configure volar properly as in https://github.com/neovim/nvim-lspconfig/blob/master/doc/configs.md#volar after "use a global TypeScript Server installation". Just change tsdk location to the proper location of volar install.
6. Now open a .vue file. You will have two lsp attached to the buffer : ts_ls and volar. All lsp features are working: i can goto definition, hover symbols, I have diagnostics, rename ... . I can fetch a location list for all symbols by running : vim.lsp.buf.document_symbol() ... . 
However, no symbols appear in the outline window when opened. I've tried Aerial and Neositter plugins, both of which encounter the same issue—LSP functions, but symbols don't display.

Vue files combine HTML/CSS and TypeScript within the same file. While I’m unsure of the exact workings, ts_ls might detect if the cursor is in an HTML or TypeScript block, using the plugin and Volar accordingly. This could be contributing to the issue.
 
I reviewed the code but lack the familiarity with the Neovim API to find where symbols are fetched from the LSP. Your plugin is excellent for displaying symbol trees, and adding support for Vue files would be amazing! 
Keep it up

**Reply by vilanele** [reply]
And also when running `:OutlineStatus`, only ts_ls is listed for lsp, volar is not. I guess volar is providing the symbols, so that might be the problem ....
